### PR TITLE
[AArch64] Correctly calculate stub_size().

### PR DESCRIPTION
**Pull request by dave-estes**
Additionally refactor extract_spoff() using a lighter-weight
instruction decoding approach in keeping with the x86_64 and PPC64
implementations.

### CODE REVIEW

File: hphp/runtime/vm/jit/service-requests.cpp [hunk 1]
```diff
@@ -254,42 +266,35 @@ FPInvOffset extract_spoff(TCA stub) {
...
-      return FPInvOffset{-(*decoder.offset / int32_t{sizeof(Cell)})};
+      auto instr = reinterpret_cast<vixl::Instruction*>(stub);
+
+      if (instr->IsAddSubImmediate()) {
+        auto const offBytes = safe_cast<int32_t>(instr->ImmAddSub());
+        always_assert((offBytes % sizeof(Cell)) == 0);
+
+        if (instr->Mask(vixl::AddSubImmediateMask) == vixl::SUB_w_imm ||
+            instr->Mask(vixl::AddSubImmediateMask) == vixl::SUB_x_imm)
+          return FPInvOffset{offBytes / int32_t{sizeof(Cell)}};
```
Comment (mxw) [inline hphp/runtime/vm/jit/service-requests.cpp]: Please no goto fails in HHVM.

Comment (aorenste) [inline hphp/runtime/vm/jit/service-requests.cpp]: Just to be a little more clear - please surround the return statement with braces so it's not a single hanging statement.

Comment (dave-estes) [inline hphp/runtime/vm/jit/service-requests.cpp]: Thanks for that. I was gonna inquire. :-) Will post an update shortly.

Comment (dave-estes) [inline hphp/runtime/vm/jit/service-requests.cpp]: Done.

**Pull request closed without merging**